dcrec/edwards/pubkey.go

```diff
@@ -32,6 +32,10 @@ func ParsePubKey(pubKeyStr []byte) (key *PublicKey, err error) {
 	if len(pubKeyStr) == 0 {
 		return nil, errors.New("pubkey string is empty")
 	}
+	if len(pubKeyStr) > PubKeyBytesLen {
```

Comment:
While this solves the original issue, the issue is still present for passed slices with len < 32.

In particular, given the encoding of pubkeys required by the eddsa RFC (https://www.rfc-editor.org/rfc/rfc8032#section-5.1.5 - item [4]) it's not safe to either append _or_ prepend bytes to the passed slice: prepending modifies the value since it's encoded as a little endian and appending causes information loss due to the oddness bit being lost (it's stored in the MSB of the last byte of the slice).

So it seems reasonable to me to test for `!=` here (instead of `>`). Note that the `copyBytes()` call below _appends_ bytes to `pubKeyStr` (which is wrong on a number of other things, but nevermind those other things), so slices with `< 32` are bungled up and the `xIsNegBytes` flag inside `encodedBytesToBigIntPoint()` will always be false.

I double checked and indeed the only place this is used in consensus codes is on the sigalt opcode (which enforces 32 byte pubkeys) so switching to `!=` is fine from that particular pov. 

Of course, external callers relying on the existing (wrong) behavior would be seriously broken, but again, as noted above non 32 byte eddsa keys are undefined anyway.

Reply:
Agreed.  Updated.